cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -1218,26 +1226,44 @@ macro(build_gtest)
         -Wno-ignored-attributes)
   endif()
 
+  if(WIN32)
+    set(GTEST_CMAKE_CXX_FLAGS ${GTEST_CMAKE_CXX_FLAGS} -DGTEST_CREATE_SHARED_LIBRARY=1)
+  endif()
+
   set(GTEST_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/googletest_ep-prefix/src/googletest_ep")
   set(GTEST_INCLUDE_DIR "${GTEST_PREFIX}/include")
+
+  if(MSVC)
+    set(_GTEST_IMPORTED_TYPE IMPORTED_IMPLIB)
+    set(_GTEST_LIBRARY_SUFFIX
+        "${CMAKE_GTEST_DEBUG_EXTENSION}${CMAKE_IMPORT_LIBRARY_SUFFIX}")
+  else()
+    set(_GTEST_IMPORTED_TYPE IMPORTED_LOCATION)
+    set(_GTEST_LIBRARY_SUFFIX
+        "${CMAKE_GTEST_DEBUG_EXTENSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
+  endif()
+
+  set(GTEST_SHARED_LIB
+      "${GTEST_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}gtest${_GTEST_LIBRARY_SUFFIX}")
+  set(GMOCK_SHARED_LIB
+      "${GTEST_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}gmock${_GTEST_LIBRARY_SUFFIX}")
   set(
-    GTEST_STATIC_LIB
-    "${GTEST_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gtest${CMAKE_GTEST_DEBUG_EXTENSION}${CMAKE_STATIC_LIBRARY_SUFFIX}"
-    )
-  set(
-    GTEST_MAIN_STATIC_LIB
-    "${GTEST_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gtest_main${CMAKE_GTEST_DEBUG_EXTENSION}${CMAKE_STATIC_LIBRARY_SUFFIX}"
+    GTEST_MAIN_SHARED_LIB
+    "${GTEST_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}gtest_main${_GTEST_LIBRARY_SUFFIX}"
     )
   set(GTEST_CMAKE_ARGS
       ${EP_COMMON_CMAKE_ARGS}
       "-DCMAKE_INSTALL_PREFIX=${GTEST_PREFIX}"
       "-DCMAKE_INSTALL_LIBDIR=lib"
-      -DCMAKE_CXX_FLAGS=${GTEST_CMAKE_CXX_FLAGS})
+      -DBUILD_SHARED_LIBS=ON
+      -DCMAKE_CXX_FLAGS="${GTEST_CMAKE_CXX_FLAGS}")
   set(GMOCK_INCLUDE_DIR "${GTEST_PREFIX}/include")
-  set(
-    GMOCK_STATIC_LIB
-    "${GTEST_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gmock${CMAKE_GTEST_DEBUG_EXTENSION}${CMAKE_STATIC_LIBRARY_SUFFIX}"
-    )
+
+  if(MSVC)
+    set(GTEST_CMAKE_ARGS ${GTEST_CMAKE_ARGS}
+        "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE=${BUILD_OUTPUT_ROOT_DIRECTORY}/Release"
+        "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG=${BUILD_OUTPUT_ROOT_DIRECTORY}/Debug")
```

```suggestion
        "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG=${BUILD_OUTPUT_ROOT_DIRECTORY}/${CMAKE_BUILD_TYPE}")
```
or maybe `BUILD_SUBDIR_NAME`? This doesn't seem like it'd work with `RelWithDebInfo`, for example

---

So MSVC adds these `Release` directories. Is it case sensitive?